Packages/CommonUI/CommonUI/Classes/View/GridView/GridView.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  GridView.swift
+//  ClearKeep
+//
+//  Created by VietAnh on 1/4/21.
+//
+
+import Foundation
+import SwiftUI
+
+public struct GridView<Content: View, T: Hashable>: View {
+	
+	private let columns: Int
+	private let hasBigItem: Bool
+	
+	// Multi-dimensional array of your list. Modified as per rendering needs.
+	private var list: [[T]] = []
+	
+	// This block you specify in 'UIGrid' is stored here
+	private let content: (T) -> Content
+	
+	public init(columns: Int, list: [T], @ViewBuilder content:@escaping (T) -> Content) {
+		self.columns = columns
+		self.content = content
+		self.hasBigItem = list.count % 2 == 1
+		self.setupList(list)
+	}
+	
+	// setupList(_:) Converts your array into multi-dimensional array.
+	private mutating func setupList(_ list: [T]) {
+		var column = 0
+		var columnIndex = 0
+		
+		for object in list {
+			if columnIndex < self.columns {
+				if columnIndex == 0 {
+					self.list.insert([object], at: column)
+					columnIndex += 1
+					
+					/// Show single item on the first row if total items is odd
+					if hasBigItem && column == 0 {
+						columnIndex = self.columns
+					}
+				} else {
+					self.list[column].append(object)
+					columnIndex += 1
+				}
+			} else {
+				column += 1
+				self.list.insert([object], at: column)
+				columnIndex = 1
+			}
+		}
+	}
+	
+	private func widthOfItem(numberItemInRow: Int) -> CGFloat {
+		if numberItemInRow > 0 {
+			return UIScreen.main.bounds.size.width / CGFloat(numberItemInRow)
+		} else {
+			return UIScreen.main.bounds.size.width/CGFloat(self.columns)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/e5e0a713c2a0f396e41d6374bb8f07759fda2936/Packages/CommonUI/CommonUI/Classes/View/GridView/GridView.swift#L60"></span>Operators should be surrounded by a single whitespace when they are being used.<br />
<code>operator_usage_whitespace</code> <code>GridView.swift:60</code></td>
    </tr>
  </tbody>
</table>